### DISCUSSION TITLE
Prioritize loop_functions over timeline_variables.

### DISCUSSION THREAD

**Discussion by kurokida** [jspsych/jsPsych#2707] [Q&A] [answered]
Hi all, 

I would like to repeat trials with the timeline variables during a predetermined time. So I wrote the loop node as follows:

```javascript
    const loop_node = {
        timeline: [trial_1],
        timeline_variables: tv,
        loop_function: function(data){
            console.log('loop!')
            if (performance.now() - base_time < phase1_time){
                return true
            } else {
                return false
            }
        }
    }
```

This does not work as I expected, the loop function is called after all the timeline variables are evaluated.
I want to make the loop function to be called for each timeline varible.

Is it possible?

**Comment by jodeleeuw** [comment, marked as answer]
Hi @kurokida,

There's no way to change when the loop function is evaluated yet. This is an interesting use case that I hadn't thought of before, so I'm glad you brought it up.

I think one solution would be to add an `on_finish` to `trial_1` that checks the time, and if the time has elapsed you can call `jsPsych.endCurrentTimeline()` ([docs](https://www.jspsych.org/7.2/reference/jspsych/#jspsychendcurrenttimeline)).

**Reply by kurokida** [reply]
Thank you for your rapid reply. The method you suggested solved my problem.

I think there are cases where the loop function should be prioritized over timeline variables. For example, stimulus image presentation or the stroop task might be repeated for a certain period of time. While these tasks are feasible in the current version of jsPsych, it's convenient if the evaluation timing of the loop function can be changed. In the new feature, it might be useful to be able to refer to the elapsed time from the start time of the loop node.
